qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferView.cpp

```diff
@@ -0,0 +1,335 @@
+
+void IETView::setSingleRebin(bool enable) {
+  m_uiForm.valRebinLow->setVisible(enable);
+  m_uiForm.valRebinWidth->setVisible(enable);
+  m_uiForm.valRebinHigh->setVisible(enable);
+}
+
+void IETView::setMultipleRebin(bool enable) { m_uiForm.valRebinString->setVisible(enable); }
+
+void IETView::setSaveEnabled(bool enable) {
+  enable = !m_outputWorkspaces.empty() ? enable : false;
+  m_uiForm.pbSave->setEnabled(enable);
+  m_uiForm.ckSaveAclimax->setEnabled(enable);
+  m_uiForm.ckSaveASCII->setEnabled(enable);
+  m_uiForm.ckSaveDaveGrp->setEnabled(enable);
+  m_uiForm.ckSaveNexus->setEnabled(enable);
+  m_uiForm.ckSaveNXSPE->setEnabled(enable);
+  m_uiForm.ckSaveSPE->setEnabled(enable);
+}
+
+void IETView::setPlotTimeIsPlotting(bool plotting) {
+  m_uiForm.pbPlotTime->setText(plotting ? "Plotting..." : "Plot");
+  setButtonsEnabled(!plotting);
+}
+
+void IETView::setFileExtensionsByName(QStringList calibrationFbSuffixes, QStringList calibrationWSSuffixes) {
+  m_uiForm.dsCalibrationFile->setFBSuffixes(calibrationFbSuffixes);
+  m_uiForm.dsCalibrationFile->setWSSuffixes(calibrationWSSuffixes);
+}
+
+void IETView::setOutputWorkspaces(std::vector<std::string> const &outputWorkspaces) {
+  m_outputWorkspaces = outputWorkspaces;
+}
+
+void IETView::setInstrumentDefault(InstrumentData const &instrumentDetails) {
+  auto const instrumentName = instrumentDetails.getInstrument();
+  auto const specMin = instrumentDetails.getDefaultSpectraMin();
+  auto const specMax = instrumentDetails.getDefaultSpectraMax();
+
+  m_uiForm.dsRunFiles->setInstrumentOverride(QString::fromStdString(instrumentName));
+
+  QStringList qens;
+  qens << "IRIS"
+       << "OSIRIS";
+  m_uiForm.spEfixed->setEnabled(qens.contains(QString::fromStdString(instrumentName)));
+
+  QStringList allowDefaultGroupingInstruments;
+  allowDefaultGroupingInstruments << "TOSCA";
+  includeExtraGroupingOption(allowDefaultGroupingInstruments.contains(QString::fromStdString(instrumentName)),
+                             "Default");
+
+  m_uiForm.spSpectraMin->setMinimum(specMin);
+  m_uiForm.spSpectraMin->setMaximum(specMax);
+  m_uiForm.spSpectraMin->setValue(specMin);
+
+  m_uiForm.spSpectraMax->setMinimum(specMin);
+  m_uiForm.spSpectraMax->setMaximum(specMax);
+  m_uiForm.spSpectraMax->setValue(specMax);
+
+  m_uiForm.spPlotTimeSpecMin->setMinimum(1);
+  m_uiForm.spPlotTimeSpecMin->setMaximum(specMax);
+  m_uiForm.spPlotTimeSpecMin->setValue(1);
+
+  m_uiForm.spPlotTimeSpecMax->setMinimum(1);
+  m_uiForm.spPlotTimeSpecMax->setMaximum(specMax);
+  m_uiForm.spPlotTimeSpecMax->setValue(1);
+
+  m_uiForm.spEfixed->setValue(instrumentDetails.getDefaultEfixed());
+
+  auto const rebinDefault = QString::fromStdString(instrumentDetails.getDefaultRebin());
+  if (!rebinDefault.isEmpty()) {
+    m_uiForm.leRebinString->setText(rebinDefault);
+    m_uiForm.ckDoNotRebin->setChecked(false);
+    auto const rbp = rebinDefault.split(",", Qt::SkipEmptyParts);
+    if (rbp.size() == 3) {
```

Comment:
I would like to see this logic in the presenter. Would it be possible and if so can we have an issue for it?

Reply:
I do like the sound of that

Reply:
I agree that setting the default values logic shouldn't be in the view. I will put an issue for that.
@RichardWaiteSTFC your suggestion sounds reasonable but my intention for writing the data classes is to make them hold only immutable data without logic. This made it easier to limit their purpose for data communication and ensure that they can't be modified

Reply:
you need to leave enough detail so someone else could pick it up

Reply:
issues raised in the roadmap issue